api/handlers/metadata.py

```diff
@@ -54,6 +57,74 @@ def get_meta(
     return metadata
 
 
+@router.get(
+    "/api/datasources/{account}/{container}/{filepath:path}/iqdata", response_class=StreamingResponse
+)
+async def get_metadata_iqdata(
+    account: str,
+    container: str,
+    filepath: str,
+    datasources_collection: Collection[DataSource] = Depends(
+        database.database.datasources_collection
+    ),
+):
+    # Create the imageURL with sasToken
+    datasource = datasources_collection.find_one(
+        {
+            "account": account,
+            "container": container,
+        }
+    )
+    if not datasource:
+        raise HTTPException(status_code=404, detail="Datasource not found")
+
+    imageURL = add_URL_sasToken(account, container, datasource["sasToken"], filepath, uiImage.IQDATA)
+    if not imageURL.get_secret_value():
+        return StreamingResponse((b'' for _ in range(1))) # return empty image
```

Comment:
I might be confused here but isnt this request to this url to get iqdata what does that had to do with image url ?